xchange-lgo/src/main/java/info/bitrich/xchangestream/lgo/LgoStreamingTradeService.java

```diff
@@ -0,0 +1,179 @@
+package info.bitrich.xchangestream.lgo;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import info.bitrich.xchangestream.core.StreamingTradeService;
+import info.bitrich.xchangestream.lgo.domain.*;
+import info.bitrich.xchangestream.lgo.dto.*;
+import info.bitrich.xchangestream.service.netty.StreamingObjectMapperHelper;
+import io.reactivex.Observable;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.*;
+import org.knowm.xchange.lgo.LgoAdapters;
+import org.knowm.xchange.lgo.dto.key.LgoKey;
+import org.knowm.xchange.lgo.dto.order.*;
+import org.knowm.xchange.lgo.service.*;
+import si.mazi.rescu.SynchronizedValueFactory;
+
+import java.io.IOException;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public class LgoStreamingTradeService implements StreamingTradeService {
+
+    private final LgoStreamingService streamingService;
+    private final LgoKeyService keyService;
+    private final LgoSignatureService signatureService;
+    private final SynchronizedValueFactory<Long> nonceFactory;
+    private final Map<CurrencyPair, LgoUserBatchSubscription> batchSubscriptions = new ConcurrentHashMap<>();
+    private Observable<LgoOrderEvent> afrSubscription;
+
+    LgoStreamingTradeService(LgoStreamingService streamingService, LgoKeyService keyService, LgoSignatureService signatureService, SynchronizedValueFactory<Long> nonceFactory) {
+        this.streamingService = streamingService;
+        this.keyService = keyService;
+        this.signatureService = signatureService;
+        this.nonceFactory = nonceFactory;
+    }
+
+    /**
+     * {@inheritDoc}
+     * First sent orders will be current open orders.
+     */
+    @Override
+    public Observable<Order> getOrderChanges(CurrencyPair currencyPair, Object... args) {
+        return getOrderBatchChanges(currencyPair)
+                .flatMap(Observable::fromIterable);
+    }
+
+    /**
+     * Get an up-to-date view of all your open orders after each LGO batch execution.
+     * First sending will be the actual open orders list.
+     */
+    public Observable<OpenOrders> getOpenOrders(CurrencyPair currencyPair) {
+        return getOrderUpdates(currencyPair)
+                .map(u -> u.getAllOpenOrders().values().stream()
+                        .filter(order -> order instanceof LimitOrder)
```

Comment:
Why the filter?  You can pass limit orders to the first parameter of `OpenOrders`' constructor, and stop orders to the second.

Reply:
It was more to deal with MarketOrder here, since we don't have Stop orders, and to my knowledge, OpenOrders doesn't support these.

Reply:
Ah, they return `MarketOrder`s? Fine then.